File: DaggerMVP/app/src/main/java/com/example/cpu10225/daggermvp/di/ActivityModule.java
package com.example.cpu10225.daggermvp.di;

import com.example.cpu10225.daggermvp.util.anotation.CustomScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by cpu10225 on 14/11/2017.
 */

@Module
public class ActivityModule {
    private final ActivityContract.View mView;

    public ActivityModule(ActivityContract.View mView) {
        this.mView = mView;
    }

    @Provides
    @CustomScope
    ActivityContract.View provideActivityContractView() {
        return mView;
    }
}

File: KotlinOnAndroid/app/src/main/java/com/example/cpu10225/kotlinonandroid/di/module/ActivityModule.kt
package com.example.cpu10225.kotlinonandroid.di.module

import com.example.cpu10225.kotlinonandroid.KotlinApp
import com.example.cpu10225.kotlinonandroid.ui.news.NewsMvpPresenter
import com.example.cpu10225.kotlinonandroid.ui.news.NewsMvpView
import com.example.cpu10225.kotlinonandroid.ui.news.NewsPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by cpu10225 on 08/12/2017.
 */

@Module
class ActivityModule() {
    @Provides
    @Singleton
    fun provideNewsMvpPresenter(presenter: NewsPresenter<NewsMvpView>): NewsMvpPresenter<NewsMvpView> = presenter
}